vllm/v1/worker/tpu_worker.py

```diff
@@ -2,350 +2,15 @@
-
-    def add_lora(self, lora_request: LoRARequest) -> bool:
-        return self.model_runner.add_lora(lora_request)
-
-    def load_model(self) -> None:
-        self.model_runner.load_model()
-
-    def update_config(self, overrides: dict[str, Any]) -> None:
-        self.model_runner.update_config(overrides)
-
-    def reload_weights(self) -> None:
-        self.model_runner.reload_weights()
-
-    def compile_or_warm_up_model(self) -> None:
-        if not self.model_config.enforce_eager:
-            self.model_runner.capture_model()
-
-        # Reset the seed to ensure that the random state is not affected by
-        # the model initialization and profiling.
-        set_random_seed(self.model_config.seed)
-
-    def reset_mm_cache(self) -> None:
-        self.model_runner.reset_mm_cache()
-
-    def get_model(self) -> nn.Module:
-        return self.model_runner.get_model()
-
-    def get_supported_tasks(self) -> tuple[SupportedTask, ...]:
-        return self.model_runner.get_supported_tasks()
-
-    def get_kv_cache_spec(self) -> dict[str, KVCacheSpec]:
-        return self.model_runner.get_kv_cache_spec()
-
-    def initialize_from_config(self, kv_cache_config: KVCacheConfig) -> None:
-        """Allocate GPU KV cache with the specified kv_cache_config."""
-        self.model_runner.initialize_kv_cache(kv_cache_config)
-
-    def check_health(self) -> None:
-        # worker will always be healthy as long as it's running.
-        return
-
-    def _init_tpu_worker_distributed_environment(
-        self,
-        vllm_config: VllmConfig,
-        rank: int,
-        distributed_init_method: str | None = None,
-        local_rank: int = -1,
-    ) -> None:
-        """Initialize the distributed environment."""
-        if self.use_spmd:
-            xr.use_spmd()
-        # NOTE(woosuk): This is just to initialize the TP group and broadcast
-        # the input objects on CPU. The all-reduce and all-gather ops on TPU
-        # are invoked by `xm.all_reduce` and `xm.all_gather` which use their
-        # own context.
-        parallel_config = vllm_config.parallel_config
-        init_distributed_environment(
-            world_size=parallel_config.world_size,
-            rank=rank,
-            local_rank=local_rank,
-            distributed_init_method=distributed_init_method or "env://",
-            backend=current_platform.dist_backend,
-        )
-        ensure_model_parallel_initialized(
-            parallel_config.tensor_parallel_size, parallel_config.pipeline_parallel_size
-        )
-
-        ensure_kv_transfer_initialized(vllm_config)
-
-    def shutdown(self) -> None:
-        self.model_runner.ensure_kv_transfer_shutdown()
-
-    def apply_model(self, fn: Callable[[nn.Module], _R]) -> _R:
-        """Apply a function on the model inside this worker."""
-        return fn(self.get_model())
-
-
 if USE_TPU_INFERENCE:
     from tpu_inference.worker.tpu_worker import TPUWorker as TpuInferenceWorker
```

Comment:
Could we add a TODO to use the hardware plugin in an official way? After that, we don't need such a tpu_workder.py file.